fleet-be/src/main/java/com/everis/training/fleet/business/fleet/boundary/VehicleResource.java

```diff
@@ -30,28 +30,44 @@ public Response listVehicles() {
     return Response.ok().entity(con.getAllVehicles()).build();
   }
 
-  @Path("free")
+  @Path("free-vehicles")
   @GET
   public Response listFreeVehicles() {
     return Response.ok().entity(con.getAllFreeVehicles()).build();
   }
 
+  @Path("{customerId}/reserve-vehicle/{vin}")
```

Comment:
Would it not be better that both "reserve-vehicle" and "finalize-reservation" would be on the same resource? I would vote for place both on VehicleResource.
And just a semantic matter, the path is {customerId}/reserve-vehicle/{vin}. Since this is the vehicleResource, the literal "vehicle" is redundant, I would prefer using only "reseve". I also observe that the path parameters are in the other way around, since the identifier is the VIN. So, a correct path would look like as {vin}/reserve/{customerId}.